### PR TITLE
mail watcher configuration using groovy script

### PR DESCRIPTION
**Pull request by tebuta**

### CODE REVIEW

File: jobs/jenkins-master/spec [hunk 1]
```diff
@@ -59,6 +60,12 @@ properties:
         username: test
         password: password
         type: UsernamePasswordCredentialImpl
+  jenkins.globalproperties.notifywhenonline:
+    description: "Email address to send notification to when node becomes available"
+    default: "monitise@service-now.com"
```
Comment (danieljamesrees) [inline jobs/jenkins-master/spec]: Doesn't this need to be configurable going by @anthonysroberts' diatribe?

File: jobs/jenkins-master/spec [hunk 2]
```diff
@@ -59,6 +60,12 @@ properties:
         username: test
         password: password
         type: UsernamePasswordCredentialImpl
+  jenkins.globalproperties.notifywhenonline:
+    description: "Email address to send notification to when node becomes available"
+    default: ""
```
Comment (cagiti) [inline jobs/jenkins-master/spec]: if the default is null, then it's possibly worth a check in the groovy script

File: jobs/jenkins-master/spec [hunk 3]
```diff
@@ -59,6 +60,12 @@ properties:
...
+  jenkins.globalproperties.notifywhenoffline:
+    description: "Email address to send notification to when node goes offline"
+    default: ""
```
Comment (cagiti) [inline jobs/jenkins-master/spec]: if the default is null, then it's possibly worth a check in the groovy script

File: jobs/jenkins-master/templates/config/configure-master-notify-global-property.groovy.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import jenkins.model.*
+import hudson.slaves.*
+import org.jenkinsci.plugins.mailwatcher.*
+
+def jenkins = Jenkins.getInstance()
+
+jenkins.getGlobalNodeProperties().add(new WatcherNodeProperty("<%= p('jenkins.globalproperties.notifywhenonline') %>","<%= p('jenkins.globalproperties.notifywhenoffline') %>"));
```
Comment (cagiti) [inline jobs/jenkins-master/templates/config/configure-master-notify-global-property.groovy.erb]: will this fail when the default value is used?

**Pull request merged**